src/libcore/num/f32.rs

```diff
@@ -205,18 +205,25 @@ impl Float for f32 {
         }
     }
 
-    /// Returns `true` if `self` is positive, including `+0.0` and
-    /// `Float::infinity()`.
+    /// Returns `true` if and only if `self` is has a positive sign, including `-0.0`, `NaN`s with
```

Comment:
I'm more surprised that it returns `true` for "`-0.0`, `NaN`s with negative sign, and negative infinity". I would not expect those cases to have a positive sign.